src/main/java/de/tum/in/www1/artemis/web/rest/metis/conversation/dtos/ConversationDTO.java

```diff
@@ -0,0 +1,163 @@
+package de.tum.in.www1.artemis.web.rest.metis.conversation.dtos;
+
+import java.time.ZonedDateTime;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import de.tum.in.www1.artemis.domain.metis.conversation.Conversation;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes({ @JsonSubTypes.Type(value = OneToOneChatDTO.class, name = "oneToOneChat"), @JsonSubTypes.Type(value = GroupChatDTO.class, name = "groupChat"),
+        @JsonSubTypes.Type(value = ChannelDTO.class, name = "channel"), })
+public class ConversationDTO {
```

Comment:
Could also be a record?

Reply:
Sadly not. Records already extend [java.lang.Record](https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/lang/Record.html). As Java does not allow multiple inheritance, records cannot extend any other class.

Reply:
I could do it with this parent class but would be weird IMO if the parent class is a record but the subclasses not.